src/Accordion/Accordion.tsx

```diff
@@ -1,174 +1,98 @@
-export type AccordionContentType = AccordionContent | Array<AccordionContent> | React.ReactNode;
-export type AccordionContent = { title?: string; desc?: string };
+import { isPrimitive } from "@sebgroup/frontend-tools/dist/isPrimitive";
+import { AccordionItemBody } from "./AccordionItemBody";
+import "./accordion.scss";
 
 export interface AccrodionListItem {
-    header: string;
-    subHeaderText?: string;
-    content?: AccordionContentType;
+    /** The header of the accordion item */
+    header: React.ReactNode;
+    /** A sub-header description rendered under the header */
+    subHeader?: React.ReactNode;
+    /** The content of the accordion item */
+    content: React.ReactNode;
 }
 
 export interface AccordionProps {
+    /** Element class name */
     className?: string;
-    customIcon?: JSX.Element;
-    customIconExpanded?: JSX.Element;
-    iconPosition?: "left" | "right";
-    iconRotation?: AccordionIconRotation;
+    /** Element id */
     id?: string;
+    /**
+     * The accordion list of items
+     * @see AccordionListItem interface
+     */
     list: Array<AccrodionListItem>;
+    /** An alternative version of the accordion */
     alternative?: boolean;
-    activeIndex?: number;
+    /** The index of an item that would be set to be expanded by default  */
+    defaultExpanded?: number;
 }
 
-const Accordion: React.FunctionComponent<AccordionProps> = (props: AccordionProps) => {
-    const collapsableRef: React.MutableRefObject<Array<React.RefObject<HTMLDivElement>>> = React.useRef(props.list.map(() => React.createRef<HTMLDivElement>()));
-    const [heightList, setHeightList] = React.useState<Array<number>>(Array(props.list.length).fill(0));
-    const [active, setActive] = React.useState<number>(null);
-    const [accordionClassName, setAccordionClassName] = React.useState<string>("custom-accordion");
-    const [itemClassName, setItemClassName] = React.useState<string>("custom-accordion");
+const Accordion: React.FC<AccordionProps> = React.memo((props: AccordionProps) => {
+    const [active, setActive] = React.useState<number>(props.defaultExpanded);
     const [idList, setIdList] = React.useState<Array<string>>([]);
+    const [id, setId] = React.useState<string>("");
+    const [className, setClassName] = React.useState<string>("");
 
+    React.useEffect(() => constructIds(), [props.list]);
+    React.useEffect(() => setActive(props.defaultExpanded), [props.defaultExpanded]);
+    React.useEffect(() => setId(props.id || randomId("accordion-")), [props.id]);
     React.useEffect(() => {
-        constructIds();
-    }, [props.list]);
-    React.useEffect(() => constructClassName(), [props.className, props.alternative]);
-    React.useEffect(() => constructItemClassName(), [props.iconPosition, props.customIconExpanded, props.iconRotation]);
-    React.useEffect(() => toggle(props.activeIndex), [props.activeIndex]);
+        let cn: string = "seb accordion";
```

Comment:
this classname is meant to be like this? or `seb-accordion`?

Reply:
`accordion` is predefined from bootstrap so I don't want to change it. However, to customize it a little bit and wanted to differentiate it from the rest. Like if you chose to build your own custom accordion, our styles will not have any effects on it. Only components that require a bit of customization over Bootstrap will have `seb` class appended to the main parent.

In short, yes, it is meant to be like this 😄